joblib/_parallel_backends.py

```diff
@@ -295,7 +295,7 @@ def __init__(self, batch):
         # arguments in memory
         self.results = batch()
 
-    def get(self):
+    def get(self, timeout=None):
```

Comment:
I think this is required, otherwise the call to `job.get` would fail in sequential mode:

https://github.com/joblib/joblib/pull/340/files#diff-12c1bd53fbe4c0df54e8edee36f2d2e9L671

However that makes me realize: wouldn't we want to have the timeout work also in sequential mode (e.g. when `n_jobs == 1`)? That seems complicated to implement (we would not to manage a timeout thread) and I don't think people would really need that feature anyway.